docker/build/Dockerfile

```diff
@@ -18,7 +18,8 @@ COPY --from=alpine/helm:3.12.2 /usr/bin/helm /usr/local/bin/
 
 COPY --from=golangci/golangci-lint:v1.51.2 /usr/bin/golangci-lint /usr/local/bin/
 
-RUN wget -O /usr/local/bin/kind https://github.com/kubernetes-sigs/kind/releases/download/v0.18.0/kind-linux-amd64 \
+RUN ARCH=$(uname -m | sed -e 's/aarch64/arm64/') && \
+    wget -O /usr/local/bin/kind https://github.com/kubernetes-sigs/kind/releases/download/v0.18.0/kind-linux-${ARCH} \
```

Comment:
> This resolves into `https://github.com/kubernetes-sigs/kind/releases/download/v0.18.0/kind-linux-x86_64` for `amd64`, so it doesn't exist. Maybe getting it from the input argument instead of `uname` would be a better idea.

Sorry about that!

I updated to use the `TARGETPLATFORM` buildx argument. Buildx was already set up in the GH workflow for the build container, and I also added the `platform` parameter there to build for both platforms.